server/plugin/command.go

```diff
@@ -501,6 +562,173 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
+	}
+
+	hookDetails, _, err := p.GetHook(ctx, githubClient, owner, repo, *githubHook.ID)
+	if err != nil {
+		return err.Error()
+	}
+
+	hookURL := baseURL + owner
+	if repo != "" {
+		hookURL += "/" + repo
+	}
+	hookURL += "/settings/hooks/"
+	txt := "Webhook Created Successfully \n"
+	hookID := strconv.Itoa(int(*githubHook.ID))
+	txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hookDetails.Events, " , "))
+	return txt
+}
+func (p *Plugin) handleWebhookList(ctx context.Context, parameters []string, baseURL string, githubClient *github.Client) string {
+	if len(parameters) != 1 {
+		return "Unknown action, please use `/github help` to see all actions available."
+	}
+
+	owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+	var githubHooks []*github.Hook
+	var githubResponse *github.Response
+	var err error
+	opt := &github.ListOptions{
+		PerPage: 10,
+	}
+	if repo == "" {
+		for {
+			githubHooks, githubResponse, err = githubClient.Organizations.ListHooks(ctx, owner, opt)
+			if err != nil {
+				return err.Error()
+			}
+			if githubResponse.NextPage == 0 {
+				break
+			}
+			opt.Page = githubResponse.NextPage
+		}
+	} else {
+		for {
+			githubHooks, githubResponse, err = githubClient.Repositories.ListHooks(ctx, owner, repo, opt)
+			if err != nil {
+				return err.Error()
+			}
+			if githubResponse.NextPage == 0 {
+				break
+			}
+			opt.Page = githubResponse.NextPage
+		}
+	}
+
+	var txt string
+	if len(githubHooks) == 0 {
+		if repo == "" {
+			txt = fmt.Sprintf("There are currently no GitHub webhooks created for the [%s](https://github.com/%s) org.", owner, owner)
+		} else {
+			txt = fmt.Sprintf("There are currently no GitHub webhooks created for the [%s](https://github.com/%s) org or the [%s/%s](https://github.com/%s/%s) repository.", owner, owner, owner, repo, owner, repo)
+		}
+	} else {
+		txt = "### Webhooks in this Repository\n"
+	}
+
+	for _, hook := range githubHooks {
+		hookID := strconv.Itoa(int(*hook.ID))
+		hookURL := baseURL + owner
+		if repo != "" {
+			hookURL += "/" + repo
+		}
+		hookURL += "/settings/hooks/"
+		txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hook.Events, " , "))
+	}
+
+	return txt
+}
+func (p *Plugin) handlewebhook(_ *plugin.Context, args *model.CommandArgs, parameters []string, userInfo *GitHubUserInfo) string {
+	if len(parameters) == 0 {
+		return "Invalid webhook command. Available command are `add` and `list` ."
+	}
+	command := parameters[0]
+	parameters = parameters[1:]
+	ctx := context.Background()
+	githubClient := p.githubConnectUser(ctx, userInfo)
+	baseURL := p.getBaseURL()
+	switch command {
+	case add:
+		return p.handleWebhookAdd(ctx, parameters, baseURL, githubClient)
+	case list:
+		return p.handleWebhookList(ctx, parameters, baseURL, githubClient)
+	}
+	return "Invalid action, only `add` and `list` commands are supported."
+}
+func (p *Plugin) CheckOptionsValid(options []string) error {
```

Comment:
Functions should always have a space between them. Not sure how CI didn't catch this

```suggestion
}

func (p *Plugin) CheckOptionsValid(options []string) error {
```